sktime/forecasting/ets.py

```diff
@@ -299,14 +305,27 @@ def _fit(error, trend, seasonal, damped):
                 )
             )
 
+            # Store IC values for each model in a list
+            # Ignore infinite likelihood models if ignore_inf_ic is True
+            _ic_list = []
+            for result in _fitted_results:
+                ic = getattr(result[1], self.information_criterion)
+                if self.ignore_inf_ic and np.isinf(ic):
+                    _ic_list.append(np.nan)
+                else:
+                    _ic_list.append(ic)
+
             # Select best model based on information criterion
-            # Get index of best model
-            _index = np.argmin(
-                [
-                    getattr(result[1], self.information_criterion)
-                    for result in _fitted_results
-                ]
-            )
+            if np.all(np.isnan(_ic_list)):
```

Comment:
What happens if `self.ignore_inf_ic` is True, then `_ic_list` would be empty right? So should we write:
```suggestion
            if np.all(np.isnan(_ic_list)) or len(_ic_list) == 0:
```

Reply:
Yes, that's right. I missed this completely.